src/components/application_manager/src/commands/mobile/on_button_press_notification.cc

```diff
@@ -106,20 +106,22 @@ void OnButtonPressNotification::Run() {
       continue;
     }
 
-    //Send ButtonPress notification only in HMI_FULL or HMI_LIMITED mode
+    // Send ButtonPress notification only in HMI_FULL or HMI_LIMITED mode
     if ((mobile_api::HMILevel::HMI_FULL != subscribed_app->hmi_level()) &&
         (mobile_api::HMILevel::HMI_LIMITED != subscribed_app->hmi_level())) {
-      LOG4CXX_WARN_EXT(logger_, "OnButtonPress in HMI_BACKGROUND or NONE");
+      LOG4CXX_WARN_EXT(logger_, "OnButtonPress notification is allowed only"
+                             << "in FULL or LIMITED hmi level when AppID is present");
       continue;
     }
 
-    //Send ButtonPress notification for OK button only in HMI_FULL mode
-    if ((static_cast<uint32_t>(mobile_apis::ButtonName::OK) == btn_id) &&
-        (mobile_api::HMILevel::HMI_FULL != subscribed_app->hmi_level())) {
-      continue;
+    // if "appID" is present  send it to Named app , only if its FULL or
+    // LIMITED
+    if (app && app->app_id()) {
+      SendButtonPress(subscribed_app);
+    } else if (subscribed_app->IsFullscreen()) {
+      // if No "appID" - send it FULL apps only.
+      SendButtonPress(subscribed_app);
```

Comment:
@OHerasym, seems there are same actions in "if" and in "else if". Can it be just one "if"? (is there important side effects in IsFullscreen?)